test/integration/driver.go

```diff
@@ -122,75 +129,33 @@ func main() {
 	}
 }
 
-func setupManager() error {
+func deploy(svcName, svcType, namespace string) {
 	// write template
-	f, err := os.Create(managerYaml)
-	if err != nil {
-		return err
-	}
-	w := bufio.NewWriter(f)
-
-	if err := write("test/integration/manager.yaml.tmpl", map[string]string{
-		"hub": hub,
-		"tag": tag,
-	}, w); err != nil {
-		return err
-	}
-
-	if err := w.Flush(); err != nil {
-		return err
-	}
-	if err := f.Close(); err != nil {
-		return err
-	}
-
-	return run("kubectl apply -f " + managerYaml + " -n " + namespace)
-}
-
-func setupSimpleApp() error {
-	// write template
-	f, err := os.Create(simpleAppYaml)
-	if err != nil {
-		return err
-	}
-	w := bufio.NewWriter(f)
-
-	if err := write("test/integration/http-service.yaml.tmpl", map[string]string{
-		"hub":   hub,
-		"tag":   tag,
-		"name":  "a",
-		"port1": "8080",
-		"port2": "80",
-	}, w); err != nil {
-		return err
-	}
-
-	if err := write("test/integration/http-service.yaml.tmpl", map[string]string{
-		"hub":   hub,
-		"tag":   tag,
-		"name":  "b",
-		"port1": "80",
-		"port2": "8000",
-	}, w); err != nil {
-		return err
+	svcConfigFile := svcName + "-" + svcType + ".yaml"
+	var w *bufio.Writer
+	if f, err := os.Create(svcConfigFile); err != nil {
+		log.Fatal(err)
+	} else {
+		defer f.Close()
+		w = bufio.NewWriter(f)
 	}
 
-	if err := write("test/integration/external-services.yaml.tmpl", map[string]string{
+	if err:= write("test/integration/"+svcType+".yaml.tmpl", map[string]string{
 		"hub":       hub,
 		"tag":       tag,
 		"namespace": namespace,
+		"name":      svcName,
+		"port1":     "8080",
```

Comment:
we need 8080/80 and 80/8080 combination - there are a lot of edge cases and this test is reducing coverage.